lib/src/executable/watch.dart

```diff
@@ -267,7 +267,7 @@ class _Watcher {
           p.setExtension(p.relative(source, from: sourceDir), '.css'));
 
       // Don't compile ".css" files to their own locations.
-      if (destination != source) return destination;
+      if (p.relative(destination) != p.relative(source)) return destination;
```

Comment:
```suggestion
      if (!p.equals(destination, source)) return destination;
```

[`p.equals()`](https://pub.dev/documentation/path/latest/path/equals.html) is the best way to determine whether two paths actually refer to the same location.